src/main/java/nextstep/security/authorization/AuthorityAuthorizationManager.java

```diff
@@ -1,16 +1,18 @@
 package nextstep.security.authorization;
 
 import nextstep.security.authentication.Authentication;
+import nextstep.security.authorization.hierarchy.RoleHierarchy;
 
 import java.util.Set;
 
 public class AuthorityAuthorizationManager<T> implements AuthorizationManager<T> {
-    private final AuthoritiesAuthorizationManager delegate = new AuthoritiesAuthorizationManager();
+    private final AuthoritiesAuthorizationManager delegate;
 
     private final String authority;
 
-    public AuthorityAuthorizationManager(String authority) {
+    public AuthorityAuthorizationManager(String authority, RoleHierarchy roleHierarchy) {
```

Comment:
Spring Security의 AuthoritiesAuthorizationManager는 String... authorities를 사용하여 다중 권한을 처리할 수 있도록 설계되었습니다.
가변인자 활용을 통해 AuthorityAuthorizationManager가 더 유연하게 동작할 수 있도록 해보면 어떨까요? 😄 
```suggestion
    public AuthorityAuthorizationManager(RoleHierarchy roleHierarchy, String... authorities) {
```